### PR TITLE
Feature/implement see more views

### PR DESCRIPTION
**Pull request by jaisonv**
Implement view to open when selecting "See more" on home view.

### CODE REVIEW

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift [hunk 1]
```diff
@@ -177,6 +189,10 @@ extension HomeViewController: SectionHeaderDelegate {
         let sectionType = presenter.sectionType(for: sectionIndex)
         let dataSource = presenter.dataSource(for: sectionType)
 
-        #warning("push to specific section here")
+        let drinkChartPresenter = DrinkChartPresenter(drinks: dataSource)
+        let drinkChartCollectionViewController = DrinkChartCollectionViewController(presenter: drinkChartPresenter)
+        drinkChartCollectionViewController.title = sectionType.title
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: Wouldn't it be better to inject this like presenter

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift [hunk 2]
```diff
@@ -177,6 +189,10 @@ extension HomeViewController: SectionHeaderDelegate {
         let sectionType = presenter.sectionType(for: sectionIndex)
         let dataSource = presenter.dataSource(for: sectionType)
 
-        #warning("push to specific section here")
+        let drinkChartPresenter = DrinkChartPresenter(drinks: dataSource)
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: Can you change to `DrinkListPresenter` and `DrinkListViewController`

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/Views/DrinkChartCell.swift [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+//  DrinkChartCell.swift
+//  CocktailRecipes
+//
+//  Created by Jaison Vieira on 10/17/20.
+//  Copyright © 2020 Eric Rado. All rights reserved.
+//
+
+import UIKit
+
+final class DrinkChartCell: UICollectionViewCell {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/Charts/Views/DrinkChartCell.swift]: I think we can reuse `SmallDrinkCell` and just add `rankLabel` to it

Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/Charts/Views/DrinkChartCell.swift]: It's done already. This is an outdated file. It was removed on my last commit.

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift (hunk 2 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: image

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift (hunk 1 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: Done.

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+//  DrinkListPresenter.swift
+//  CocktailRecipes
+//
+//  Created by Jaison Vieira on 10/17/20.
+//  Copyright © 2020 Eric Rado. All rights reserved.
+//
+
+import Foundation
+
+struct DrinkListPresenter {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift]: I have been using classes for presenters

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListViewController.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+//  DrinkListViewController.swift
+//  CocktailRecipes
+//
+//  Created by Jaison Vieira on 10/17/20.
+//  Copyright © 2020 Eric Rado. All rights reserved.
+//
+
+import UIKit
+
+class DrinkListViewController: UICollectionViewController {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListViewController.swift]: final

File: CocktailRecipes/CocktailRecipes/Features/Home/Views/SmallDrinkCell.swift [hunk 6]
```diff
@@ -9,57 +9,79 @@
...
 
-	func configure(image: UIImage?, text: String) {
+    func configure(image: UIImage?, text: String, rank: Int?) {
 		imageView.image = image
-		label.text = text
+        titleLabel.text = text
+
+        if let rank = rank {
+            rankLabel.text = "\(rank)"
+            rankLabel.isHidden = false
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/Views/SmallDrinkCell.swift]: I think we can remove this and the else and combine it to one line `rankLabel.isHidden = rank == nil`

Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/Views/SmallDrinkCell.swift]: done

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListViewController.swift (hunk 5 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListViewController.swift]: done

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift (hunk 4 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift]: Done.

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift [hunk 7]
```diff
@@ -176,6 +188,10 @@ extension HomeViewController: SectionHeaderDelegate {
         let sectionType = presenter.sectionType(for: sectionIndex)
         let dataSource = presenter.dataSource(for: sectionType)
 
-        #warning("push to specific section here")
+        let drinkChartPresenter = DrinkListPresenter(drinks: dataSource)
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: rename the properties and the one below it too

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+//  DrinkListPresenter.swift
+//  CocktailRecipes
+//
+//  Created by Jaison Vieira on 10/17/20.
+//  Copyright © 2020 Eric Rado. All rights reserved.
+//
+
+import Foundation
+
+class DrinkListPresenter {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift]: final

File: CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift (hunk 7 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/HomeViewController.swift]: Done.

File: CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift (hunk 8 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipes/Features/Home/Charts/DrinkListPresenter.swift]: Done.

**Pull request merged**